scripts/deps

```diff
@@ -273,6 +287,46 @@ def remove_tree(path):
         os.remove(path)
 
 
+def install_to_system(deps, architecture, system):
+    """Install binaries from deps to /usr/local/bin/"""
+    install_dir = "/usr/local/bin"
+
+    # Check if we have write permission
+    if not os.access(install_dir, os.W_OK):
+        print(f"Error: No write permission to {install_dir}. You may need to run with sudo.")
+        return False
+
+    installed = []
+    for dep in deps:
+        if not dep.matches_platform(architecture, system):
+            continue
+
+        for binary in dep.get_binaries():
+            source_path = dep.get_binary_target_path(binary)
```

Comment:
`deps.get_binary_source_path` maybe is better. It's the same in the end but currently this reads like a typo (`source_path = dep.get_binary_target_path(binary)`).